include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -151,31 +151,34 @@ struct __composite_kernel_name
 // and instantiate template with variadic non-type template parameters.
 // This approach is required to get reliable work group size when kernel is unnamed
 #if _ONEDPL_BUILT_IN_STABLE_NAME_PRESENT
-template <typename _Tp>
+template <typename _KernelName, typename _Tp>
 class __kernel_name_composer
 {
     static constexpr auto __name = __builtin_sycl_unique_stable_name(_Tp);
     static constexpr ::std::size_t __name_size = __builtin_strlen(__name);
 
     template <::std::size_t... _Is>
-    static __composite_kernel_name<__name[_Is]...>
+    static __composite_kernel_name<_KernelName, __name[_Is]...>
     __compose_kernel_name(::std::index_sequence<_Is...>);
 
   public:
     using type = decltype(__compose_kernel_name(::std::make_index_sequence<__name_size>{}));
 };
+
+template <typename _KernelName, typename _Tp>
+using __kernel_name_composer_t = typename __kernel_name_composer<_KernelName, _Tp>::type;
 #endif // _ONEDPL_BUILT_IN_STABLE_NAME_PRESENT
 
-template <template <typename...> class _BaseName, typename _CustomName, typename... _Args>
+template <template <typename...> class _BaseName, typename _KernelName, typename _CustomName, typename... _Args>
 using __kernel_name_generator =
 #if __SYCL_UNNAMED_LAMBDA__
-    typename ::std::conditional<_HasDefaultName<_CustomName>::value,
+    ::std::conditional_t<_HasDefaultName<_CustomName>::value,
 #    if _ONEDPL_BUILT_IN_STABLE_NAME_PRESENT
-                                typename __kernel_name_composer<_BaseName<_CustomName, _Args...>>::type,
+                                __kernel_name_composer_t<_KernelName, _BaseName<_CustomName, _Args...>>,
 #    else // _ONEDPL_BUILT_IN_STABLE_NAME_PRESENT
                                 _BaseName<_CustomName, _Args...>,
 #    endif
```

Comment:
That is correct.

On the other hand, that comment forces me thinking that our name looks ugly because we shorten it for the use-case of converting to characters. So, the question is do we believe that we still support such an old DPC++ compiler without necessary built-in or any other compiler beside DPC++ that has support for unnamed lambda?

If the answer on the question above is Yes, then the next question would be about use-case frequency.

@akukanov, Do we believe that the use-case I've described above deserves attention? If yes, then we have completely unreadable name (for the user; for us it's ugly but can be read) and might potentially rethink the approach for shorten kernel names.

Reply:
The question is no more. Adding extra symbols to name doesn't affect compilation time. True, it does the name a little bit longer from Tools perspective but also makes our code more readable and easier to support.